hubspot/modules.go

```diff
@@ -1,40 +1,21 @@
 package hubspot
 
 import (
-	"fmt"
+	"github.com/amp-labs/connectors/common/paramsbuilder"
 )
 
-type APIModule struct {
-	Label   string // e.g. "crm"
-	Version string // e.g. "v3"
-}
-
 // ModuleCRM is the module used for accessing standard CRM objects.
-var ModuleCRM = APIModule{ // nolint: gochecknoglobals
+var ModuleCRM = paramsbuilder.APIModule{ // nolint: gochecknoglobals
 	Label:   "crm",
 	Version: "v3",
 }
 
 // ModuleEmpty is used for proxying requests through.
-var ModuleEmpty = APIModule{ // nolint: gochecknoglobals
+var ModuleEmpty = paramsbuilder.APIModule{ // nolint: gochecknoglobals
 	Label:   "",
 	Version: "",
 }
 
 // supportedModules represents currently working and supported modules within the Hubspot connector.
 // Any added module should be appended added here.
-var supportedModules = []APIModule{ModuleCRM} // nolint: gochecknoglobals
-
-func (a APIModule) String() string {
-	return fmt.Sprintf("%s/%s", a.Label, a.Version)
-}
-
-func supportsModule(module string) bool {
-	for _, mod := range supportedModules {
-		if module == mod.String() {
-			return true
-		}
-	}
-
-	return false
-}
+var supportedModules = []paramsbuilder.APIModule{ModuleCRM} // nolint: gochecknoglobals
```

Comment:
Would this mean that an empty module initialization of Hubspot would error out?

Reply:
Thats the understanding I got from the way it is implemented right now. It is refactored in such a way where any connector that will have support of module option must be one of the listed, where we enforce the default.

Empty Module is not used in connectors and I have just checked that there is one usage in `server`. So what happens there is it will fallback to ModuleCRM.

Reply:
We need it to be able to initialize with an empty module so that it can work as a proxy connector. Can you please add a fix for this? The fallback should be NoModule.

Reply:
The PR has a commit with this fix: https://github.com/amp-labs/connectors/pull/615/commits/1e631f2632e1873f2e11a89085ddb6b897118708